src/server/api/routers/user.ts

```diff
@@ -32,6 +32,48 @@ export const userRouter = createTRPCRouter({
     return await db.user.findMany();
   }),
 
+  createUser: protectedProcedure
+    .input(
+      z.object({
+        name: z.string().min(1),
+        email: z.string().email(),
+        role: z.string().min(1),
+      }),
+    )
+    .mutation(async ({ input }) => {
+      try {
+        const existingUser = await db.user.findUnique({
+          where: {
+            email: input.email,
+          },
+        });
+        if (existingUser) {
+          throw new TRPCError({
+            code: "CONFLICT",
+            message: "A user with this email already exists.",
+          });
+        }
+        return await db.user.create({
```

Comment:
We should test this to make sure this works with google once we make google work